rotkehlchen/chain/ethereum/decoding/kyber/decoder.py

```diff
@@ -0,0 +1,73 @@
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple
+
+from rotkehlchen.accounting.structures import (
+    HistoryBaseEntry,
+    HistoryEventSubType,
+    HistoryEventType,
+)
+from rotkehlchen.chain.ethereum.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.ethereum.decoding.structures import ActionItem
+from rotkehlchen.chain.ethereum.structures import EthereumTxReceiptLog
+from rotkehlchen.chain.ethereum.types import string_to_ethereum_address
+from rotkehlchen.chain.ethereum.utils import ethaddress_to_asset
+from rotkehlchen.types import ChecksumEthAddress, EthereumTransaction
+from rotkehlchen.user_messages import MessagesAggregator
+from rotkehlchen.utils.misc import hex_or_bytes_to_address
+
+if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.ethereum.manager import EthereumManager
+
+TOKEN_SWAP = b"\x18I\xbdj\x03\n\x1b\xca(\xb847\xfd=\xe9o='\xa5\xd1r\xfa~\x9cx\xe7\xb6\x14h\x92\x8a9"  # noqa: E501
```

Comment:
Which event is this? I don't see an even log with the name TokenSwap here: https://etherscan.io/tx/0x615c28b9867c786d7365a7d60f90a68b95cc2acd025fb0f96e22134e8dc2602f#eventlog